lib/internal/process/task_queues.js

```diff
@@ -158,13 +159,18 @@ const defaultMicrotaskResourceOpts = { requireManualDestroy: true };
 function queueMicrotask(callback) {
   validateFunction(callback, 'callback');
 
-  const asyncResource = new AsyncResource(
-    'Microtask',
-    defaultMicrotaskResourceOpts,
-  );
-  asyncResource.callback = callback;
-
-  enqueueMicrotask(FunctionPrototypeBind(runMicrotask, asyncResource));
+  const contextFrame = AsyncContextFrame.current();
+  if (contextFrame || getHookArrays()[0].length > 0) {
```

Comment:
Is `enabledHooksExist()` not sufficient for this check? Why do you need to retrieve the actual hooks array?

Reply:
There as a similar discussion in a similar PR https://github.com/nodejs/node/pull/59873#issuecomment-3304866589
It returns always `true` because it [checks async_hook_fields[kCheck]](https://github.com/nodejs/node/blob/e28656a6172c35d88a472e33894a11f0e9f98eee/lib/internal/async_hooks.js#L479-L485) which is initialized to 1 [here](https://github.com/nodejs/node/blob/e28656a6172c35d88a472e33894a11f0e9f98eee/src/env.cc#L1758-L1763).

We could think about removing this as indicated in the TODO there or add a separate function which returns the right thing.